### PR TITLE
adding conventional changelog action & auto-release

### PR DESCRIPTION
**Pull request by Aryan-Deshpande**
### 👨‍💻 Changes proposed
- addition of release.yaml file that implements changelog and auto-release

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Releases
+
+on:
+  push:
+    branches: [ main ]
+
```
Comment (Pradumnasaraf) [inline .github/workflows/release.yaml:6]:
```suggestion
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        uses: TriPSs/conventional-changelog-action@v3
+        with:
+          gihtub-token: ${{ secrets.CHANGELOG_RELEASE }}
+
+      - name: create release
+        uses: actions/create-release@v1
+        if: ${{ steps.changelog.skipped == 'false' }}
+
+        env:
+          GITHUB_TOKEN: ${{ secrets.CHANGELOG_RELEASE }}
```
Comment (Pradumnasaraf) [inline .github/workflows/release.yaml:27]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.github_token }}
```

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          tag_name: ${{ steps.changelog.outputs.tag }}
+          release_name: ${{ steps.changeog.outputs.tag }}
+          body: ${{ steps.changelog.outputs.clean_changelog }}
```
Comment (Pradumnasaraf) [inline .github/workflows/release.yaml:31]:
```suggestion
          body: ${{ steps.changelog.outputs.clean_changelog }}

```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: conventional Changelog Action
+
+        id: changelog
```
Comment (Pradumnasaraf) [inline .github/workflows/release.yaml:17]:
```suggestion
      - name: conventional Changelog Action
        id: changelog
```

**Review by Pradumnasaraf — CHANGES_REQUESTED** [review]
Hey,

#### Changes required

- Fix the indentation issue.
- Change the GitHub token to default one `secrets.github_token`

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by Aryan-Deshpande** [comment]
hey, @Pradumnasaraf applied recommended changes, thanks for the help!

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+name: Releases
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
```
Comment (ykdojo) [inline .github/workflows/release.yaml:9]: where does it run exactly - ubuntu?

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (ykdojo) [inline .github/workflows/release.yaml:12]: what does this line mean?

**Review by ykdojo — COMMENTED** [review]
Started reviewing this, and I have a few questions about this.

1. What's the whole point of doing it?

2. There are many lines in this yaml file - did you write all of them manually, or does it come from a template?

**Comment by Pradumnasaraf** [comment]
> Started reviewing this, and I have a few questions about this.
> 
> 1. What's the whole point of doing it?
> 2. There are many lines in this yaml file - did you write all of them manually, or does it come from a template?

Hey, Yk. Answering to questions: 

1. It's a Github workflow which will run a GitHub Action that create auto-releases for you by using conventional commits.
2. It's a template. We also use in our EddieHub projects.



__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

**Comment by Pradumnasaraf** [comment]
Hey, @ykdojo. Answering questions: 

1. It's a Github workflow which will run a GitHub Action that creates auto-releases for you by using conventional commits.
2. It's a template. We also use it in our EddieHub projects.


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

File: .github/workflows/release.yaml (hunk 5 above)
Comment (Pradumnasaraf) [reply .github/workflows/release.yaml:9]: Github Action runs then internally.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Tags:** Efficient

File: .github/workflows/release.yaml (hunk 6 above)
Comment (Pradumnasaraf) [reply .github/workflows/release.yaml:12]: It uses the checkout action by GitHub for going inside the repo and checking it - https://github.com/actions/checkout

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by Pradumnasaraf** [comment]
> hey, @Pradumnasaraf applied recommended changes, thanks for the help!

Thank you for the changes.

**Review by Pradumnasaraf — APPROVED** [review]
Looks good to merge!

**Comment by ykdojo** [comment]
Cool, thanks a lot for the clarification @Pradumnasaraf

**Pull request merged**